src/data/set/intervals/monotone.lean

```diff
@@ -177,3 +178,46 @@ calc (⋃ x, Ioo (f x) (g x)) = (⋃ x, Ioi (f x)) ∩ ⋃ x, Iio (g x) :
 ... = Ioi a ∩ Iio b : congr_arg2 (∩) ha.Union_Ioi_eq hb.Union_Iio_eq
 
 end Union
+
+section succ_order
+
+open order
+
+variables {α : Type*} [partial_order α] [order_bot α]
+  [succ_order α] [is_succ_archimedean α] [no_max_order α] {n : α} {φ : α → α}
+
+lemma strict_mono_on.Iic_id_le (hφ : strict_mono_on φ (set.Iic n)) :
+  ∀ m ≤ n, m ≤ φ m :=
```

Comment:
This should be true without `no_max_order`: if the preivious `m` is not a maximal element, then `h : succ m = m`, and `rwa [h]` closes the goal.